lib/linklist.c

```diff
@@ -296,3 +297,36 @@ void list_add_list(struct list *l, struct list *m)
 	for (n = listhead(m); n; n = listnextnode(n))
 		listnode_add(l, n->data);
 }
+
+struct list *list_dup(struct list *l)
```

Comment:
Shouldn't this inherit any cmp or delete functions?

Reply:
yep